src/main/java/com/microsoft/sqlserver/jdbc/SQLServerXAConnection.java

```diff
@@ -29,13 +29,16 @@ public final class SQLServerXAConnection extends SQLServerPooledConnection imple
      * transactional processing to the application. That app server is the one who should restrict commit/rollback on
      * the connections it issues to applications, not the driver. These instances can and must commit/rollback
      */
-    private SQLServerXAResource XAResource;
-    
-    /** physical connection */
+    private volatile SQLServerXAResource XAResource;
+
+    /**
+     * physical connection
+     */
```

Comment:
Can be a single line comment like line 39 below.

Reply:
I'll commit a revert.